src/Processors/ResizeProcessor.cpp

```diff
@@ -350,10 +355,8 @@ IProcessor::Status StrictResizeProcessor::prepare(const PortNumbers & updated_in
 
         auto & waiting_output = output_ports[input_with_data.waiting_output];
 
-        if (waiting_output.status == OutputStatus::NotActive)
-            throw Exception(ErrorCodes::LOGICAL_ERROR, "Invalid status NotActive for associated output");
```

Comment:
What is the reason for removing this error check?

Reply:
The output port which already associated with an input port may transition to a NotActive state if the output port is updated and abandoned chunks are pushed into it.

> What is the reason for removing this error check?

Reply:
I've discussed this issue with the original author of this code and we decided that these changes are raising many questions and the solution is not really clear. Could you please write a unit test regarding this issue if you have some free time?